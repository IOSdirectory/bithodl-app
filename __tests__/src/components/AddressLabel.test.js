import React from 'react';
import { Clipboard } from 'react-native';
import renderer from 'react-test-renderer';
import AddressLabel from '../../../src/components/AddressLabel';

describe('AddressLabel', () => {
  let fakeAddress;

  beforeEach(() => {
    fakeAddress = '2N6oSTcdmjmTPEQ1sLTBye8HCvbM2fyL2HM';
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <AddressLabel address={fakeAddress} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('#_copyToClipboard()', () => {
    let addressLabel;

    beforeEach(() => {
      Clipboard.setString.mockClear();

      // Mock setState(). Should probably use some utility to handle this.
      // E.g. <https://github.com/airbnb/enzyme>
      AddressLabel.prototype.setState = jest.fn();

      addressLabel = new AddressLabel({
        address: fakeAddress
      });
    });

    it('accepts no arguments', () => {
      const length = addressLabel._copyToClipboard.length;
      expect(length).toBe(0);
    });

    it('copies the address to the clipboard', () => {
      addressLabel._copyToClipboard();

      expect(Clipboard.setString).toHaveBeenCalledTimes(1);
      expect(Clipboard.setString).toHaveBeenCalledWith(fakeAddress);
    });
  });
});
