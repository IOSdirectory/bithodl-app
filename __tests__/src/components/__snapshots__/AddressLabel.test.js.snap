// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressLabel renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#F9F9F9",
        "borderRadius": 3,
        "flexDirection": "row",
        "overflow": "hidden",
        "padding": 8,
      },
      undefined,
    ]
  }
  testID={undefined}
>
  <View
    style={
      Object {
        "borderRightColor": "#E4E4E4",
        "borderRightWidth": 1,
        "marginRight": 7,
        "paddingLeft": 1,
        "paddingRight": 7,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "color": "#7C7C7C",
          },
        ]
      }
    >
      â‚¿
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#504F63",
          "fontFamily": "Avenir Next",
          "fontWeight": "400",
        },
        Object {
          "color": "#7C7C7C",
          "fontFamily": "Menlo-Regular",
          "fontSize": 12,
          "lineHeight": 13,
          "padding": 2,
        },
      ]
    }
  >
    2N6oSTcdmjmTPEQ1sLTBye8HCvbM2fyL2HM
  </Text>
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F9F9F9",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "color": "#7C7C7C",
            "fontSize": 11,
            "fontWeight": "600",
          },
        ]
      }
    >
      Copied
    </Text>
  </View>
</View>
`;
